oauthproxy.go

```diff
@@ -1001,12 +1001,12 @@ func (p *OAuthProxy) addHeadersForProxying(rw http.ResponseWriter, req *http.Req
 			req.Header.Del("X-Forwarded-Preferred-Username")
 		}
 
+		req.Header.Del("X-Forwarded-Groups")
```

Comment:
Good catch -- we'll need to fix this.

@JoelSpeed Another one for you to deconflict from the inflight headers PR that is deleting this code block

Reply:
I imagine this change will also need to apply to the `X-Auth-Request-Groups` as well?

Reply:
What about clearing all auth headers before doing any session processing?

```golang
func (p *OAuthProxy) Proxy(rw http.ResponseWriter, req *http.Request) {
	p.stripAuthHeaders(req)
	session, err := p.getAuthenticatedSession(rw, req)
```
Are there reasons to keep auth headers set downstream?

Reply:
Oh, I almost forgot -- This might be completely resolved in this related PR once it potentilly merges: https://github.com/oauth2-proxy/oauth2-proxy/pull/799

Reply:
Thanks for looking into this and pointing me to #826 . It looks like the new middleware will specifically address this issue by deleting existing auth headers as long as `header.PreserveRequestValue` is false.

Reply:
Indeed that is the intention, any header that is configured to be added by the proxy should be cleared unless specifically requested to be preserved, are you happy to close this and await that feature being live?

Reply:
Yes, that sounds good. Closing this and corresponding issue #877. Thanks again.